### ISSUE TITLE
Configure doesn't fail when enable-mca-no-build mca and component names are invalid

### ISSUE THREAD

**Issue by Akshay-Venkatesh** [open-mpi/ompi#7481]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
master, 4.0.2

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

git, tarball

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.

```
./configure ... --enable-mca-no-build=btl-utc ...
```

## Details of the problem

We expected configure to fail because there was a typo `btl-utc` where there should've been `btl-uct`. But because there's no check on validity of mca and instance name [here](https://github.com/open-mpi/ompi/blob/ceb52126a8064ae41011df7baec737b08edf1a6b/config/opal_mca.m4#L101), there was no failure. Would it be possible to get configure to fail if names are invalid? Can avoid some problems later on.

cc @jsquyres @luitjens

**Comment by jsquyres** [comment]
Agreed: this is a bug.  We don't validate the framework/component names that are given to `--enable-mca-no-build`.  We also don't validate the names given to `--enable-mca-dso`, `--enable-mca-static` or `--enable-mca-direct`.

PR's to fix the issue would be appreciated. 😄

**Comment by jsquyres** [comment]
FYI: this is where `--enable-mca-no-build` is created and parsed: https://github.com/open-mpi/ompi/blob/af1ec9a59487ffe0e61efac9ef3688c4a0ad3a00/config/opal_mca.m4#L58-L118

Might not be too hard to check and see if the various directories exist.

**Comment by bwbarrett** [comment]
> Might not be too hard to check and see if the various directories exist.

You shouldn't use directories to check for existence.  There's not a 1:1 mapping between directories and component.  You'd have to wait until after the MCA run and use the list of found valid components / frameworks.  A little late, but that's how the system works now.
